pkg/test/environment.go

```diff
@@ -133,10 +142,16 @@ func NewEnvironment(options ...option.Function[EnvironmentOptions]) *Environment
 
 	_ = lo.Must(environment.Start())
 
+	// Apply any config overrides
+	config := rest.CopyConfig(environment.Config)
```

Comment:
nit: Why copy? Why not just directly mutate?

Reply:
No reason, we can directly mutate.